code/datums/elements/beauty.dm

```diff
@@ -0,0 +1,69 @@
+/**
+ * Beauty element; makes the indoor area the parent is in prettier or uglier depending on the beauty var value.
+ * Clean and well decorated areas lead to positive moodlets for passerbies;
+ * Shabbier, dirtier ones lead to negative moodlets EXCLUSIVE to characters with the snob quirk.
+ */
+/datum/element/beauty
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	var/beauty = 0
+	/**
+	  * Assoc list of atoms as keys and number of time the same element instance has been attached to them as assoc value.
+	  * So things don't get odd with same-valued yet dissimilar beauty modifiers being added to the same atom.
+	  */
+	var/beauty_counter = list()
+
+/datum/element/beauty/Attach(datum/target, _beauty)
```

Comment:
Rename `_beauty` to `beauty`, and change `beauty = _beauty` to `src.beauty = beauty`. This lets us used named arguments.